src/cli/cli.cpp

```diff
@@ -485,17 +332,19 @@ void Interpreter::ProcessHelp(uint8_t aArgsLength, char *aArgs[])
     OT_UNUSED_VARIABLE(aArgsLength);
     OT_UNUSED_VARIABLE(aArgs);
 
+    static_assert(IsArraySorted(sCommands, OT_ARRAY_LENGTH(sCommands)), "Command list is not sorted");
```

Comment:
Can/should we apply these optimizations to sub-commands (e.g. `dataset`)?

Reply:
Good idea. I think we should. Will add this later.

I am also considering (for future) to add a bunch of helper methods in `CmdLineParser` (arg lookup based on a table, string match, parse as number, etc). This way they can be shared/used by CLI and factory diags.